pennylane/templates/layers/particle_conserving_u1.py

```diff
@@ -0,0 +1,251 @@
+
+    # decomposition of C-PhaseShift(2*phi) gate
+    PhaseShift(-phi, wires=m)
+    CNOT(wires=wires)
+    PhaseShift(phi, wires=m)
+    CNOT(wires=wires)
+    PhaseShift(-phi, wires=n)
+
+
+def u1_ex_gate(phi, theta, wires=None):
+    r"""Implement the two-qubit exchange gate :math:`U_{1,\mathrm{ex}}` proposed
+    in `arXiv:1805.04340 <https://arxiv.org/abs/1805.04340>`_ to build
+    a hardware-efficient particle-conserving VQE ansatz for quantum chemistry
+    simulations.
+
+    Args:
+        phi (float): angle entering the unitary :math:`U_A(\phi)`
+        theta (float): angle entering the rotation :math:`R(0, 2\theta, 0)`
+        wires (list[Wires]): the two wires ``n`` and ``m`` the circuit acts on
+    """
+
+    # C-UA(phi)
+    decompose_ua(phi, wires=wires)
+
+    qml.CZ(wires=wires[::-1])
+    qml.CRot(0, 2 * theta, 0, wires=wires[::-1])
+
+    # C-UA(-phi)
+    decompose_ua(-phi, wires=wires)
+
+
+@template
+def ParticleConservingU1(weights, wires, init_state=None):
+    r"""Implements the heuristic VQE ansatz for quantum chemistry simulations using the
+    particle-conserving gate :math:`U_{1,\mathrm{ex}}` proposed by Barkoutsos *et al.* in
+    `arXiv:1805.04340 <https://arxiv.org/abs/1805.04340>`_.
+
+    This template prepares :math:`N`-qubit trial states by applying :math:`D` layers of the
+    entangler block :math:`U_\mathrm{ent}(\vec{\phi}, \vec{\theta})` to the Hartree-Fock
+    state
+
+    .. math::
+
+        \vert \Psi(\vec{\phi}, \vec{\theta}) \rangle = \hat{U}^{(D)}_\mathrm{ent}(\vec{\phi}_D,
+        \vec{\theta}_D) \dots \hat{U}^{(2)}_\mathrm{ent}(\vec{\phi}_2, \vec{\theta}_2)
+        \hat{U}^{(1)}_\mathrm{ent}(\vec{\phi}_1, \vec{\theta}_1) \vert \mathrm{HF}\rangle.
+
+    The circuit implementing the entangler blocks is shown in the figure below:
+
+    |
+
+    .. figure:: ../../_static/templates/layers/particle_conserving_u1.png
+        :align: center
+        :width: 50%
+        :target: javascript:void(0);
+
+    |
+
+    The repeated units across several qubits are shown in dotted boxes. Each layer
+    contains :math:`N-1` particle-conserving two-parameter exchange gates
+    :math:`U_{1,\mathrm{ex}}(\phi, \theta)` that act on pairs of nearest neighbors qubits.
+    The unitary matrix representing :math:`U_{1,\mathrm{ex}}(\phi, \theta)`
+    is given by (see `arXiv:1805.04340 <https://arxiv.org/abs/1805.04340>`_),
+
+    .. math::
+
+        U_{1, \mathrm{ex}}(\phi, \theta) = \left(\begin{array}{cccc}
+        1 & 0 & 0 & 0 \\
+        0 & \mathrm{cos}(\theta) & e^{i\phi} \mathrm{sin}(\theta) & 0 \\
+        0 & e^{-i\phi} \mathrm{sin}(\theta) & -\mathrm{cos}(\theta) & 0 \\
+        0 & 0 & 0 & 1 \\
+        \end{array}\right).
+
+    The figure below shows the circuit decomposing :math:`U_{1, \mathrm{ex}}` in
+    elementary gates. The Pauli matrix :math:`\sigma_z` and single-qubit rotation :math:`R(0, 2 \theta, 0)`
+    apply the Pauli Z operator and an arbitrary rotation on the qubit ``n``,
```

Comment:
> The Pauli matrix :math:`\sigma_z` and single-qubit rotation :math:`R(0, 2 \theta, 0)`
    apply the Pauli Z operator and an arbitrary rotation on the qubit ``n``,

Think it's worth noting that in the circuit these are applied with qubit `m` being the control qubit